ormar/models/newbasemodel.py

```diff
@@ -144,15 +144,16 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:  # type: ignore
         new_kwargs, through_tmp_dict = self._process_kwargs(kwargs)
 
         if not pk_only:
-            self.__pydantic_validator__.validate_python(
+            values = self.__pydantic_validator__.validate_python(
                 new_kwargs, self_instance=self  # type: ignore
-            )
+            ).__dict__
+            fields_set = set(values.keys())
```

Comment:
I am almost sure it's initializing self (so values and fields_set) in ` self.__pydantic_validator__.validate_python` as it worked in tests w/o your change (that's why you pass self), now you do this 2 times (with this setattr below)